### PR TITLE
Improves sleepy pen logging

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Improves the logging of the sleepy pen to show the reagents that were in the pen on injection.

## Why It's Good For The Game
Sleepy pens are a nasty way to dump 50u of whatever into someone, and the logs don't clearly show what exactly was transferred. This seems like an oversight.

## Images of changes
image

### CODE REVIEW

File: code/modules/paperwork/pen.dm [hunk 1]
```diff
@@ -110,6 +110,9 @@
 /obj/item/pen/sleepy
 	container_type = OPENCONTAINER
 	origin_tech = "engineering=4;syndicate=2"
+	var/transfer_amount = 50
```
Comment (hal9000PR) [inline code/modules/paperwork/pen.dm:113]: why make this a variable if the user cannot vary it? unless I am missing something?

Comment (lewcc) [reply code/modules/paperwork/pen.dm:113]: Possible admemes. I was thinking along the lines of the changes I made for allowing VV edits with the RSG.

File: code/modules/paperwork/pen.dm [hunk 2]
```diff
@@ -119,18 +122,28 @@
...
-	reagents.add_reagent("ketamine", 100)
+	create_reagents(reagent_amount)
+	for(var/key in starting_reagents)
+		reagents.add_reagent(key, starting_reagents[key])
 
+/obj/item/pen/sleepy/vv_edit_var(var_name, var_value)
+	. = ..()
+	switch(var_name)
+		if("reagent_amount")
+			reagents.maximum_volume = var_value
```
Comment (AffectedArc07) [inline code/modules/paperwork/pen.dm:146]: Yeah I aint having a var exist just as a VV callback.

People can edit the thing.

File: code/modules/paperwork/pen.dm (hunk 1 above)
Comment (AffectedArc07) [inline code/modules/paperwork/pen.dm:113]: Not a fan of vars just for the sake of adminbus shenanigans.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by lewcc** [comment]
Sounds good, I'll remove those tweaks 👍

File: code/modules/paperwork/pen.dm [hunk 3]
```diff
@@ -119,17 +121,22 @@
 	if(!M.can_inject(user, TRUE))
 		return
 	var/transfered = 0
+	var/contained = list()
+	for(var/R in reagents.reagent_list)
+		var/datum/reagent/reagent = R
+		contained += "[reagent.volume]u [reagent]"
```
Comment (hal9000PR) [inline code/modules/paperwork/pen.dm:127]:
```suggestion
		contained += "[round(reagent.volume, 0.01)]u [reagent]"
```
just so we dont get any of those nasty super long decimals.

File: code/modules/paperwork/pen.dm [hunk 4]
```diff
@@ -119,17 +121,22 @@
...
 	return TRUE
 
 
 /obj/item/pen/sleepy/Initialize(mapload)
 	. = ..()
-	create_reagents(100)
-	reagents.add_reagent("ketamine", 100)
+	create_reagents(reagent_amount)
+	for(var/key in starting_reagents)
+		reagents.add_reagent(key, starting_reagents[key])
```
Comment (AffectedArc07) [inline code/modules/paperwork/pen.dm:139]:
```suggestion
	create_reagents(100)
	reagents.add_reagent("ketamine", 100)
```

File: code/modules/paperwork/pen.dm [hunk 5]
```diff
@@ -110,6 +110,8 @@
 /obj/item/pen/sleepy
 	container_type = OPENCONTAINER
 	origin_tech = "engineering=4;syndicate=2"
+	var/reagent_amount = 100
+	var/starting_reagents = list("ketamine" = 100)
```
Comment (AffectedArc07) [inline code/modules/paperwork/pen.dm:114]:
```suggestion
```

File: code/modules/paperwork/pen.dm [hunk 6]
```diff
@@ -119,17 +121,22 @@
 	if(!M.can_inject(user, TRUE))
 		return
 	var/transfered = 0
+	var/contained = list()
+	for(var/R in reagents.reagent_list)
+		var/datum/reagent/reagent = R
+		contained += "[round(reagent.volume, 0.01)]u [reagent]"
 	if(reagents.total_volume && M.reagents)
 		transfered = reagents.trans_to(M, 50)
 	to_chat(user, "<span class='warning'>You sneakily stab [M] with the pen.</span>")
```
Comment (AffectedArc07) [inline code/modules/paperwork/pen.dm:130]:
```suggestion
	var/contained = list()

	for(var/R in reagents.reagent_list)
		var/datum/reagent/reagent = R
		contained += "[round(reagent.volume, 0.01)]u [reagent]"

	if(reagents.total_volume && M.reagents)
		transfered = reagents.trans_to(M, 50)

	to_chat(user, "<span class='warning'>You sneakily stab [M] with the pen.</span>")
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**